chalk-solve/src/solve/slg/implementation.rs

```diff
@@ -199,6 +200,34 @@ impl context::InferenceTable<SlgContext, SlgContext> for TruncatingInferenceTabl
     fn cannot_prove(&self) -> Goal {
         Goal::CannotProve(())
     }
+
+    // Used by: logic
+    fn next_subgoal_index(&mut self, ex_clause: &ExClause<SlgContext>) -> usize {
+        ex_clause
+            .subgoals
+            .iter()
+            .enumerate()
+            .max_by(|(_, lit1), (_, lit2)| match (lit1, lit2) {
+                (Literal::Positive(_), Literal::Negative(_)) => Ordering::Greater,
+                (Literal::Negative(_), Literal::Positive(_)) => Ordering::Less,
+                (Literal::Negative(_), Literal::Negative(_)) => Ordering::Equal,
+                (
+                    Literal::Positive(InEnvironment { goal: goal1, .. }),
+                    Literal::Positive(InEnvironment { goal: goal2, .. }),
+                ) => {
+                    // Handle `?U: Trait` goals last
```

Comment:
But this looks a bit confusing. It seems to just be deprioritizing *any* trait goal. I think we'll want to be more selective. How about we land the PR with a very simple function that **just** returns `strand.ex_clause.subgoals.len() - 1`, and move the FIXME to here? =)

Reply:
That could work. We can just make a note in #80 that we still need to figure out the specifics of how best to select the next subgoal.